### PR TITLE
Cleanup XbConvert as a whole: Less magic, more mapping, more logging of unsupported features

### PR DESCRIPTION
**Pull request by LukeUsher**

### CODE REVIEW

**Comment by PatrickvL** [comment]
Please compare this with https://github.com/PatrickvL/Dxbx/blob/master/Source/Delphi/src/DxbxKrnl/EmuD3D8/uConvert.pas too

**Comment by LukeUsher** [comment]
@PatrickvL this is primarily a cleanup PR, and additions of previously unsupported Xbox extensions are outside the scope of this work. 

Other mappings (for example D3DRS_BLENDCOLOR => D3DRS_BLENDFACTOR  and D3D D3DBLEND_CONSTANTCOLOR => D3DBLEND_BLENDFACTOR) will in a later PR, once a proper test case is found and behavior is confirmed.

File: src/core/hle/D3D8/XbConvert.h [hunk 1]
```diff
@@ -141,11 +143,26 @@ inline D3DBLENDOP EmuXB2PC_D3DBLENDOP(X_D3DBLENDOP Value)
...
+        case 0x308: return D3DBLEND_SRCALPHASAT;
+        // Unsupported (on host) Xbox Extensions
+        // TODO: Find a way to implement these: Xbox sets a constant blend colour using D3DRS_BLENDCOLOR
+        case 0x8001: EmuLogEx(LOG_PREFIX_D3DCVT, LOG_LEVEL::WARNING, "Unsupported Xbox D3DBlend Extension: D3DBLEND_CONSTANTCOLOR"); return D3DBLEND_ONE;
+        case 0x8002: EmuLogEx(LOG_PREFIX_D3DCVT, LOG_LEVEL::WARNING, "Unsupported Xbox D3DBlend Extension: D3DBLEND_INVCONSTANTCOLOR "); return D3DBLEND_ONE;
+        case 0x8003: EmuLogEx(LOG_PREFIX_D3DCVT, LOG_LEVEL::WARNING, "Unsupported Xbox D3DBlend Extension: D3DBLEND_CONSTANTALPHA"); return D3DBLEND_ONE;
+        case 0x8004: EmuLogEx(LOG_PREFIX_D3DCVT, LOG_LEVEL::WARNING, "Unsupported Xbox D3DBlend Extension: D3DBLEND_INVCONSTANTALPHA"); return D3DBLEND_ONE;
+    }
 
     EmuLogEx(LOG_PREFIX_D3DCVT, LOG_LEVEL::WARNING, "Unknown Xbox D3DBLEND Extension (0x%.08X)", Value);
```
[the remark this answers is not in the document]
Comment (LukeUsher) [reply src/core/hle/D3D8/XbConvert.h]: Why? There is no change for double logging as all cases return a value, it doesn't fall through.

[the remark this answers is not in the document]
Comment (PatrickvL) [reply src/core/hle/D3D8/XbConvert.h]: Darn. You're right - I overlooked the return statements at the end of the cases that call EmuLogEx! I'll resolve this remark....

[the remark this answers is not in the document]
Comment (LukeUsher) [reply src/core/hle/D3D8/XbConvert.h]: Thanks

**Pull request merged**